crates/red_knot_python_semantic/src/types.rs

```diff
@@ -297,6 +299,10 @@ impl<'db> Type<'db> {
                 // TODO defer to Type::Instance(<bytes from typeshed>).member
                 Type::Unknown
             }
+            Type::EllipsisType => {
+                // TODO: attribute lookup on Ellipsis type
+                Type::Unknown
+            }
```

Comment:
I'm not sure this needs to be its own variant in the `Type` enum... it's conceptually just an instance of `types.EllipsisType`. `types.EllipsisType` was only exposed on Python 3.10+, but typeshed already provides a nice little compatibility workaround for us here: <https://github.com/python/typeshed/blob/f58dac1d62d33bb2255b762783d06463c40f5065/stdlib/builtins.pyi#L1849-L1865>

(`builtins.Ellipsis` is the same object as `...`)

```pycon
>>> ...
Ellipsis
>>> Ellipsis
Ellipsis
>>> Ellipsis is ...
True
```

Reply:
One interesting way in which `...` _is_ special is that it's a singleton: it is the _only_ instance of `types.EllipsisType` that there is, or that there could be. But that can be thought of as just a sealed type that has exactly one member, and we've yet to implement a generalised way of handling sealed types. (We'll need them to tackle enums in Python!) See https://github.com/astral-sh/ruff/issues/12694.